doc/process/PreReleaseChecklist.md

```diff
@@ -0,0 +1,61 @@
+# Pre Release Checklist
+
+This documents the steps to be taken prior to making a new release of the
+nav2 stack.
+
+## Summary
+1. `Ensure all dependencies are listed in the package.xml files` by doing a
+build of all of ROS2, dependencies, and navigation 2 in one workspace.
+
+2. `Ensure all dependencies are released.` by using rosdep to pull in dependencies instead of building them ourselves.
+
+3. `Ensure the test suite passes`
+
+## Detailed Steps
+
+### Ensure all dependencies are listed in the package.xml files
+
+We want to ensure that every package has a complete list of its dependencies
+in the `package.xml` file. This can be done by not sourcing any ros `setup.bash` files. Instead we need to build everything as one big repo.
+
+There is a docker file to do that, so run
+
+```bash
+sudo docker build -t nav2:crystal_full  --build-arg http_proxy=http://myproxy.example.com:80  --build-arg https_proxy=http://myproxy.example.com:80 --build-arg ROS2_BRANCH=crystal -f Dockerfile.full_ros_build ./
```

Comment:
Did you get this command and the other one swapped? This one says ROS2_BRANCH=crystal, which I think should be in the next example?

Reply:
Both dockerfiles need a ROS2_BRANCH argument now. It represents the upstream branch we are targeting or verifying against.